### ISSUE TITLE
Assigning Nested Bytecache Does not Lead to Buffer Update

### ISSUE THREAD

**Issue by osanj** [osanj/lava#1]
If one assigns buffer values not in the top most level, the buffer object flags for `fresh_bytez` is never set. For example:

```
buffer["someArray"] = [1, 1, 1]
# data will be pushed to the shader
```

```
buffer["someArray"][0] = 1
buffer["someArray"][1] = 1
buffer["someArray"][2] = 1
# data will not be pushed to the shader
```

**Comment by osanj** [comment]
Workarounds:
- for `StagedBuffer` set `staged_buffer.buffer_cpu.fresh_cache = True`
- for `BufferCPU` call `flush()`

**Comment by osanj** [comment]
Implemented by #3

**Comment by osanj** [comment]
PR #3
